core/scratch_block_comment.js

```diff
@@ -0,0 +1,404 @@
+    this.updateColour();
+  } else {
+    // Dispose of the bubble.
+    this.bubble_.dispose();
+    this.bubble_ = null;
+    this.textarea_ = null;
+    this.foreignObject_ = null;
+    this.label_ = null;
+  }
+  // Restore the bubble stats after the visibility switch.
+  this.setText(text);
+  this.setBubbleSize(size.width, size.height);
+};
+
+/**
+ * Toggle the minimization state of this comment.
+ * @private
+ */
+Blockly.ScratchBlockComment.prototype.toggleMinimize_ = function() {
+  this.setMinimized(!this.isMinimized_);
+};
+
+/**
+ * Set the minimized state for this comment.
+ * @param {boolean} minimize Whether the comment should be minimized
+ */
+Blockly.ScratchBlockComment.prototype.setMinimized = function(minimize) {
+  if (this.isMinimized_ == minimize) return;
+  this.isMinimized_ = minimize;
+  if (minimize) {
+    this.bubble_.setMinimized(true, this.getLabelText());
+    this.setBubbleSize(Blockly.ScratchBlockComment.MINIMIZE_WIDTH,
+        Blockly.ScratchBubble.TOP_BAR_HEIGHT);
+    // Note we are not updating this.width_ or this.height_ here
+    // because we want to keep track of the width/height of the
+    // maximized comment
+  } else {
+    this.bubble_.setMinimized(false);
+    this.setText(this.text_);
+    this.setBubbleSize(this.width_, this.height_);
+  }
+};
+
+/**
+ * Size this comment's bubble.
+ * @param {number} width Width of the bubble.
+ * @param {number} height Height of the bubble.
+ */
+Blockly.Comment.prototype.setBubbleSize = function(width, height) {
+  if (this.bubble_) {
+    if (this.isMinimized_) {
+      this.bubble_.setBubbleSize(width, Blockly.ScratchBubble.TOP_BAR_HEIGHT);
+    } else {
+      this.bubble_.setBubbleSize(width, height);
+      this.width_ = width;
+      this.height_ = height;
+    }
+  } else {
+    this.width_ = width;
+    this.height_ = height;
+  }
+};
+
+/**
+ * Get the truncated text for this comment to display in the minimized
+ * top bar.
+ * @return {string} The truncated comment text
+ */
+Blockly.ScratchBlockComment.prototype.getLabelText = function() {
+  if (this.text_.length > Blockly.ScratchBlockComment.MAX_LABEL_LENGTH) {
+    return this.text_.slice(0, Blockly.ScratchBlockComment.MAX_LABEL_LENGTH) + '\u2026';
+  } else {
+    return this.text_;
+  }
+};
+
+/**
+ * Set this comment's text.
+ * @param {string} text Comment text.
+ */
+Blockly.Comment.prototype.setText = function(text) {
+  if (this.text_ != text) {
+    Blockly.Events.fire(new Blockly.Events.BlockChange(
+        this.block_, 'comment', null, this.text_, text));
+    this.text_ = text;
+  }
+  if (this.textarea_) {
+    this.textarea_.value = text;
+  }
+};
+
+/**
+ * Move this comment to a position given x and y coordinates.
+ * @param {number} x The x-coordinate
+ * @param {number} y The y-coordinate
+ */
+Blockly.ScratchBlockComment.prototype.moveTo = function(x, y) {
+  if (this.bubble_) {
+    this.bubble_.moveTo(x, y);
+  }
+  this.x_ = x;
```

Comment:
Everywhere else you're using `this.x` and `this.y`.  Here and in the next function you're using `this.x_` and `this.y_`.

Reply:
Eek! Good catch.